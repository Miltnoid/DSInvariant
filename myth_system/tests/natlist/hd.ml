type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let hd : list -> nat |>
  { [] => 0
  | [0] => 0
  | [1] => 1
  } = ?
