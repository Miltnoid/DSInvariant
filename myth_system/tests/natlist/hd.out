(* Generated Wed Dec 31 11:16:29 2014 *)
let hd : list -> nat =
  fun (l1:list) -> match l1 with
                     | Nil -> O
                     | Cons (n1, l2) -> n1
;;
